api/v1alpha1/api_webhook.go

```diff
@@ -146,6 +149,39 @@ func (a *APIValidator) InjectDecoder(d *admission.Decoder) error {
 	return nil
 }
 
+func (a *APIValidator) PathAlreadyDeployed(ctx context.Context, fleet *EnvoyFleetID, newApi API) error {
+	var apiObjs APIList
+	parser := spec.NewParser(nil)
+	newApiSpec, err := parser.ParseFromReader(strings.NewReader(newApi.Spec.Spec))
+	if err != nil {
+		return err
+	}
+
+	// Get all API objects with this fleet field set
+	if err := a.Client.List(ctx, &apiObjs, &client.ListOptions{}); err != nil {
+		return fmt.Errorf("failure querying for the deployed APIs: %w", err)
+	}
+
+	// filter out apis are in the process of deletion
+	for _, api := range apiObjs.Items {
+		if api.Spec.Fleet.Name == fleet.Name && api.Spec.Fleet.Namespace == fleet.Namespace {
```

Comment:
You can invert this and turn it into a guard clause with a `continue`

Reply:
I'll just combine the statement above